ludwig/api.py

```diff
@@ -585,6 +585,7 @@ def on_epoch_end(self, trainer, progress_tracker, save_path):
                 update_config_with_metadata(self.config_obj, training_set_metadata)
                 logger.info("Warnings and other logs:")
                 self.model = LudwigModel.create_model(self.config_obj, random_seed=random_seed)
+                update_config_with_model(self.config_obj, self.model)
```

Comment:
before this line i would add a comment explaining why we need to update the config after the model has been created